### PR TITLE
chore: install and configure TypeScript

### PR DESCRIPTION
**Pull request by WesSouza**
This PR installs and configures TypeScript across our use cases.

This:
- Configures TypeScript and dependencies
- Configures Rollup to use TypeScript via `esbuild` and `rollup-plugin-dts
- Configure Storybook so docs are generated from types
- Configure ESLint
- Configure Jest using `ts-jest`
- Fix or ignore all ESLint issues
- Fix a failing test
- Converts `Anchor` to TypeScript

So far this is an incremental approach that allows JS and TS files to coexist.

I haven't checked if the output of `esbuild` is equivalent to the prior `babel` usage. I also haven't checked if we can remove babel from our explicit dependencies.

### CODE REVIEW

File: src/common/themes/peggysPastels.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
  * https://www.deviantart.com/tpenguinltg/art/Peggy-s-Pastels-505540096
  */
 
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
Comment (WesSouza) [inline src/common/themes/peggysPastels.js:5]: I don't think this is actually working, `../utils/index.js` is not exported as a file on the package relative to the themes export:

https://www.runpkg.com/?react95@3.12.1/dist/

Comment (arturbien) [reply src/common/themes/peggysPastels.js:5]: Great catch, I tried importing this theme in this sandbox and it indeed is not working: https://codesandbox.io/s/react95-template-urwo3s?file=/src/index.js

Comment (arturbien) [reply src/common/themes/peggysPastels.js:5]: fixed it in #289

**Comment by WesSouza** [comment]
@arturbien @luizbaldi I paused work on this to get some feedback from y'all:
- Can you validate if the output of Rollup using `esbuild` is equivalent to using babel and the other plugins?
- We could release TypeScript support incrementally. Would that be ok or do we want to work on a separate branch and bundle it all on a new version?

File: src/Table/Table.stories.js [hunk 2]
```diff
@@ -13,6 +13,11 @@ import {
   WindowContent
 } from 'react95';
 
+const Wrapper = styled.div`
```
Comment (arturbien) [inline src/Table/Table.stories.js:16]: Please remove these type of changes from this PR (moving Wrapper in stories to the top etc)

**Review by arturbien — CHANGES_REQUESTED** [review]

**Comment by arturbien** [comment]
@WesSouza I've noticed that prop type table is not generated for anchor component:
Screenshot 2022-07-24 at 13 30 31
 
 How it looks on prod:
Screenshot 2022-07-24 at 13 30 45

File: src/Table/Table.stories.js (hunk 2 above)
Comment (WesSouza) [reply src/Table/Table.stories.js:16]: Some components have Wrapper defined above the export already.

This was done to fix no-use-before-define errors, would you want to disable those rules?

**Comment by WesSouza** [comment]
@arturbien I tested adding more props and those appear, but not children. I wonder if that's expected from that addon.

File: src/Table/Table.stories.js (hunk 2 above)
Comment (arturbien) [reply src/Table/Table.stories.js:16]: oh ok, I didn't notice we enforced this rule. If that's the case let's keep the changes!

**Comment by arturbien** [comment]
@WesSouza let's try to collapse the commits to a fewer ones where possible and change the last commit type to `feat(anchor)` so that semantic-release releases updated version with typed anchor

**Pull request merged**